telemetry/noop_module.go

```diff
@@ -64,55 +63,55 @@ func (m *NoopTelemetryModule) GetEventMetricsAgent() modules.EventMetricsAgent {
 }
 
 func (m *NoopTelemetryModule) EmitEvent(namespace, event_name string, labels ...any) {
-	NOOP()
+	NOOP("EmitEvent", "namespace", namespace, "event_name", event_name, "labels", labels)
```

Comment:
I think using reflections here is fair game to get the function name: https://stackoverflow.com/a/7053871/768439.

We can reuse it in all the functions

Reply:
I am not sure I like it:

![image](https://user-images.githubusercontent.com/29378614/211011695-01c47416-a6be-425b-98aa-c77b6161ee5d.png)

It is "cool" that we can rely on build-time stuff like a func instead of strings but honestly I feel like it would just add needless complexity here.

It can be made prettier by creating a wrapper func that strips out the namespace and whatever was added after the function name "-fm" 🤔

Not sure if it's worth it but I am open to do it if you feel strongly about it.